.github/workflows/build.yml

```diff
@@ -44,11 +63,17 @@ jobs:
             #export CXX=clang++
             export CXX=clazy
           fi
+          if [[ "${{ matrix.container-image }}" =~ "opensuse" ]]; then
```

Comment:
Instead of hardcoding distros, check for file existance which is what ultimately matters anyway:

```
export LOGIN_DEFS_PATH=/etc/login.defs
[ -e "$LOGIN_DEFS_PATH" ] || export LOGIN_DEFS_PATH=/usr/etc/login.defs
```
 `[ -e /etc/login.defs ]

Reply:
I find it easier to read how it's currently implemented. This way it's clear it's Opensuse the  one that is different and if they ever change/fix it we will notice here and we will be able to remove unneded code

Reply:
The move to `/usr/etc` is intentional and I expect that at least some other distros follow.

Reply:
So far, nobody else uses `/usr/etc` for this purpose. Other distributions have been moving files to different locations in `/usr`. I'm also skeptical of anyone else using `/usr/etc` since `/etc` is such a bad path for config files anyway.

Reply:
:shrug: If they want to diverge, sure...

Reply:
Nobody is doing `/usr/etc` except openSUSE. In this regard, openSUSE is the divergence. Everyone else is moving things to either `/usr/share` or some other more appropriate location.

Reply:
There was some cross-distro communication but nobody else cared so we went ahead with `/usr/etc`. If someone is moving it to another location after we already use `/usr/etc` for some time now, it's not us causing the divergence.

Reply:
I could also just touch `/etc/login.defs` and make this work that way universally too.

Reply:
SDDM doesn't actually use `login.defs` during runtime at all, it's just used during buildtime to figure out the default user ID range.

Reply:
In any case, my main argument here is that checking for distro names is as bad as checking user agents in web pages. The correct way is to perform feature detection.

Reply:
Fine, I don't care enough. Fixed.